src/System.Runtime.Extensions/tests/System/Net/WebUtility.cs

```diff
@@ -2,165 +2,130 @@
-
-            Assert.Equal(input, returned);
+            Assert.Equal(expected, WebUtility.UrlDecode(encodedValue));
         }
 
-        [Fact]
-        public static void UrlDecodeFromStringNoEncodingReturnsNullIfInputIsNull()
+        public static IEnumerable<object[]> UrlEncode_TestData()
         {
-            string returned = WebUtility.UrlDecode((string)null);
-
-            Assert.Null(returned);
+            // Recent change brings function inline with RFC 3986 to return hex-encoded chars in uppercase
+            yield return new object[] { "/\\\"\tHello! \u2665?/\\\"\tWorld! \u2665?\u2665", "%2F%5C%22%09Hello!+%E2%99%A5%3F%2F%5C%22%09World!+%E2%99%A5%3F%E2%99%A5" };
+            yield return new object[] { "'", "%27" };
+            yield return new object[] { null, null };
         }
 
-        [Fact]
-        public static void UrlEncodeFromStringNoEncoding()
+        [Theory, MemberData("UrlEncode_TestData")]
+        public static void UrlEncode(string value, string expected)
         {
-            // Recent change brings function inline with RFC 3986 to return hex-encoded chars in uppercase
-            string input = "/\\\"\tHello! \u2665?/\\\"\tWorld! \u2665?\u2665";
-            string expected = "%2F%5C%22%09Hello!+%E2%99%A5%3F%2F%5C%22%09World!+%E2%99%A5%3F%E2%99%A5";
-
-            string returned = WebUtility.UrlEncode(input);
-
-            Assert.Equal(expected, returned);
+            Assert.Equal(expected, WebUtility.UrlEncode(value));
         }
 
-        [Fact]
-        public static void UrlEncodeFromStringNoEncodingReturnsNullIfInputIsNull()
+        [Theory]
+        [InlineData("'")]
+        [InlineData("http://www.microsoft.com")]
+        [InlineData("/\\\"\tHello! \u2665?/\\\"\tWorld! \u2665?\u2665")]
+        public static void UrlEncodeDecode_Roundtrip(string value)
         {
-            string returned = WebUtility.UrlEncode((string)null);
-            Assert.Null(returned);
+            string encoded = WebUtility.UrlEncode(value);
+            Assert.Equal(value, WebUtility.UrlDecode(encoded));
         }
 
-        [Fact]
-        public static void UrlEncodeToBytes_ArgumentValidation()
+        public static IEnumerable<object[]> Url_DecodeToBytes_TestData()
         {
-            Assert.Null(WebUtility.UrlEncodeToBytes(null, 0, 0));
-            Assert.Throws<ArgumentNullException>("bytes", () => WebUtility.UrlEncodeToBytes(null, 0, 1));
-            Assert.Throws<ArgumentOutOfRangeException>("offset", () => WebUtility.UrlEncodeToBytes(new byte[1], -1, 1));
-            Assert.Throws<ArgumentOutOfRangeException>("offset", () => WebUtility.UrlEncodeToBytes(new byte[1], 2, 1));
-            Assert.Throws<ArgumentOutOfRangeException>("count", () => WebUtility.UrlEncodeToBytes(new byte[1], 0, -1));
-            Assert.Throws<ArgumentOutOfRangeException>("count", () => WebUtility.UrlEncodeToBytes(new byte[1], 0, 3));
+            yield return new object[] { null, 0, 0, null };
```

Comment:
I'm not actually, but I would suggest keeping it as [Theory], as this makes the method more extendable in the future.

For example, adding a new test is as simple as adding a line of code, instead of a new test method.